3p/ampcontext.js

```diff
@@ -62,24 +110,38 @@ export class AmpContext {
    *    every time we receive a page visibility message.
    */
   observePageVisibility(callback) {
-    return this.client_.makeRequest(
-        MessageType.SEND_EMBED_STATE,
-        MessageType.EMBED_STATE,
-        callback);
+    const visibilityCallbacksLength = this.visibilityCallbacks_.length;
```

Comment:
callback might not be executed if you `makeRequest` early.

2 suggested changes:
- invoke callback immediately if there was a response.
- do `makeRequest` lazily